### DISCUSSION TITLE
Mongodb: relation removal doesn't unset parent.referencedFielter

### DISCUSSION THREAD

**Discussion by manu-urba** [mikro-orm/mikro-orm#3132] [Ideas]
**Describe the bug**
I implemented an entity with a OneToOne relation:
```ts
@Field(() => EndUser, { nullable: true })
  @OneToOne(() => EndUser, (endUser) => endUser.authUser, {
    owner: true,
    nullable: true,
  })
  endUser?: EndUser;
```

I have to implement a method which unsets this relation, with the following code
```ts
const authUser = await this.authUserRepository.findOne({ id: authUserId });
delete authUser.endUser; // or authUser.endUser = undefined;
return await this.authUserRepository.persistAndFlush(authUser);
```
This causes the following query to be fired, which sets AuthUser.endUser field as `null`
```
[query] db.getCollection('auth-user').updateMany({ _id: ObjectId('628764ff52cccc7b41960ce4') }, { '$set': { endUser: undefined } }, { session: undefined });
```

I expect it to unset it. https://www.mongodb.com/docs/manual/reference/operator/update/unset/

**Versions**

| Dependency | Version |
| - | - |
| node | v14.17.2 |
| typescript | 4.6.2 |
"@mikro-orm/core": "^5.1.4",
"@mikro-orm/mongodb": "^5.1.4",
"@mikro-orm/nestjs": "^5.0.1",

**Comment by B4nan** [comment]
Are other mongo ORMs doing this by default? I don't see much difference here, for the runtime entity there is none. I obviously do get the difference you can see when you look at the data directly.

Definitely not a bug, please don't consider everything you think should work somehow as bugs. To me this is working as expected, and what you ask might be even seen as a breaking change. So I need to hear more about the motivation before we can consider this. 

I guess I'd be open to do this optionally but I am affraid it won't be that easy change - we would have to distinguish removing props and setting them to `null`, which would probably affect other drivers, etc etc... Or we could do it always, e.g. setting the prop to `null` would behave the same, but that seems ever weirder.

**Comment by manu-urba** [comment]
Yeah I see that is not a bug, I understand your point.
The reason why it is an issue for me is because that field for me is unique, and setting it to null doesn't let me set additional ones to null.

However I would just like to ask if it's possible to do this manually, without changing Mikroorm at all. Can I send that query natively to mongo? The problem with `nativeUpdate` repository method is that it wouldn't let me use Mongo operators.

**Reply by B4nan** [reply]
You can, and you can use most of mongo operators right now even with EM, they are just not allowed on TS level, you can get around that via casting in most cases. Given we have the mongo flavour of EM, we should be able to adjust the types too, will look into it when I have a moment (edit: gave it a few minutes to prototype it and it's not that easy, TS is reporting issues with cycles :/). 

Just tried this in tests and it is indeed working just fine:

```ts
await orm.em.nativeUpdate(FooBar, {}, { $unset: { fooBar: 1 } } as any);
```

And I also tried to reproduce the original issue, and I feel like we should indeed use the `$unset` here - especially if we see the value is `undefined` and not `null`. I was hoping this works fine with sparse indexes, but apparently not. 

As a last resort, you can use `em.getCollection(Ent)` to get the native mongodb collection instance and do whatever you want with that. This works and type checks already:

```ts
await orm.em.getCollection(FooBar).updateMany({ fooBar: { $exists: true } }, { $unset: { fooBar: 1 } });
```

**Reply by B4nan** [reply]
It wasnt that hard in the end, let's hope it won't break anything, worst case we will hide it behind a configuration option.

f05981135d984b73b5f8144758ab7055c533d3cc

**Reply by manu-urba** [reply]
Thank you!
